include/vkgcDefs.h

```diff
@@ -354,6 +354,9 @@ struct PipelineOptions {
                                    ///  then linking them, when possible.  When not possible this option is ignored.
   bool disableImageResourceCheck;  ///< If set, the pipeline shader will not contain code to check and fix invalid image
                                    ///  descriptors.
+  bool enableScratchAccessBoundaryChecks; ///< If set, out of bounds guards will be inserted in the LLVM IR for OpLoads
```

Comment:
nit: consider `enableScratchAccessBoundaryChecks` --> `enableScratchAccessBoundsChecks`. I think this would be a more standard name (e.g., https://en.wikipedia.org/wiki/Bounds_checking), and easier to type